src/functional.rs

```diff
@@ -332,12 +407,12 @@ mod tests {
     }
 
     #[test]
-    fn test_delete_two_children_with_child() {
+    fn test_delete_two_children_with_grandchild() {
         let mut tree = Tree::new();
         tree = tree.insert(2, 3);
         tree = tree.insert(1, 2);
-        tree = tree.insert(0, 1);
```

Comment:
I swapped these to prevent an early balance from changing the resulting tree. As noted in the description we should probably create these trees explicitly in cases like this instead of calling `insert` over and over.